gossip/src/crds.rs

```diff
@@ -303,8 +313,19 @@ impl Crds {
                     Err(CrdsError::InsertFailed)
                 } else if matches!(route, GossipRoute::PushMessage(_)) {
                     let entry = entry.get_mut();
-                    entry.num_push_dups = entry.num_push_dups.saturating_add(1);
-                    Err(CrdsError::DuplicatePush(entry.num_push_dups))
+                    let num_push_recv = match entry.num_push_recv {
+                        Some(count) => {
+                            if count == 0 {
+                                self.stats.lock().unwrap().num_redundant_pull_responses += 1;
+                            }
+                            Some(count.saturating_add(1))
+                        }
+                        None => Some(1),
+                    };
+                    let num_push_dups = num_push_recv.unwrap_or(0).saturating_sub(1);
```

Comment:
I believe we could just use `unwrap()` here instead of `unwrap_or(0)` since `num_push_recv` will not be `None` after the `match` statement

Reply:
I think this could be written as:
```rust
if entry.num_push_recv == Some(0) {
   self.stats.lock().unwrap().num_redundant_pull_responses += 1;
}
let num_push_dups = entry.num_push_recv.unwrap_or_default();
entry.num_push_recv = Some(num_push_dups.saturating_add(1));
```